x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcAssert.java

```diff
@@ -269,11 +268,11 @@ else if (type == Types.DOUBLE) {
                     } else if (type == Types.FLOAT) {
                         assertEquals(msg, (float) expectedObject, (float) actualObject, lenientFloatingNumbers ? 1f : 0.0f);
                     } else if (type == Types.OTHER) {
-                        if (expectedObject instanceof Geometry && actualObject instanceof String) {
+                        if (actualObject instanceof org.elasticsearch.geo.geometry.Geometry) {
                             // We need to convert the actual object to Geometry for comparision
```

Comment:
The comment is leftover from the previous implementation. Before we had string in actual and JTS Geometry in expected. With this change I am trying to reduce dependency on JTS and H2 in this code, so instead of converting everything to H2's JTS Geometry I am converting it to our geometry. I will update the comment. Good catch!